### PR TITLE
[ADD] cms_security

### PR DESCRIPTION
**Pull request by simahawk**
Full explanation in [README](https://github.com/simahawk/website-cms/blob/11-add-cms_security/cms_security/README.rst).

### CODE REVIEW

File: cms_security/models/security.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+                record = env['ir.rule'].create(values)
+                utils.add_xmlid(env, record, xmlid)
+                logger.info('Created default record rule for %s: %s',
+                            self._name, values['name'])
+
+    @api.model
+    def _setup_complete(self):     # pylint: disable=R8110
+        """Override to handle automatic security."""
+        super()._setup_complete()
+        # TODO: this code runs 2 times on every reboot.
```
Comment (simahawk) [inline cms_security/models/security.py:87]: @sbidoul @guewen can you give me your insights about this?

Probably a possible solution would be to add an API method to be called by on pre/post hooks on modules installation (modules that rely on this feature) but I'd like to keep this automatic and setup everything when a new model is registered.

Comment (guewen) [reply cms_security/models/security.py:87]: It seems to me that `_setup_*` methods normally "initialize" the state of the models but don't have any side effect on the database. That could explain why it's called 2 times. Modifications in database are done in `_auto_init` and `init` called at addons upgrade / install. 

```python
    def init(self):
        """ This method is called after :meth:`~._auto_init`, and may be
            overridden to create or modify a model's database schema.
        """
        pass
```

What would be missing with an override of `init`?

Comment (guewen) [reply cms_security/models/security.py:87]: > but I'd like to keep this automatic and setup everything when a new model is registered.

Like when you create a new model from the UI? If so what about calling the initialization in `IrModuleModule.create`?

Comment (simahawk) [reply cms_security/models/security.py:87]: not just from the UI,  I'd like to catch both UI and code setup

Comment (simahawk) [reply cms_security/models/security.py:87]: @guewen I've just give a run w/ `init`: same result. It's called twice and apparently there's no way to state that everything is already in place. Seems like the only good candidate is `ir.model` create, which very likely should be called just once.

File: cms_security/README.rst [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+It provides a ``cms.security.mixin``
+that gives your models the following security behavior:
+
+* only owner can edit or delete
+* only owner can view if not published
+* only published items are visible by other users or anonymous users
+
+Additionally:
+
+* you can grant view/edit permissions by using ``read_group_ids`` and ``write_group_ids``
```
Comment (gurneyalex) [inline cms_security/README.rst:23]: could you explain this a bit more in the README? Maybe in an example below?

File: cms_security/utils.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    rr3 = 'allowed groups can view (read_group_ids)'
+    rr4 = 'allowed groups can write (write_group_ids)'
+    values = [{
+        '_xmlid': make_xmlid(prefix, rr1),
+        'name': '{} {}'.format(prefix, rr1),
+        'domain_force': """
+            ['|',
+             ('website_published', '=', True),
+             '&', ('create_uid','=',user.id),
+                  ('website_published', '=', False)
```
Comment (gurneyalex) [inline cms_security/utils.py:70]: Could this not be written as:
```
[
'|',
 ('website_published', '=', True),
 ('create_uid','=',user.id),
]
```

File: cms_security/utils.py [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+        """,
+        'active': True,
+        'perm_read': True,
+        'perm_write': False,
+        'perm_create': False,
+        'perm_unlink': False,
+        'model_id': model_id,
+        'groups': [
+            (4, env.ref('base.group_portal').id),
+            (4, env.ref('base.group_public').id),
```
Comment (gurneyalex) [inline cms_security/utils.py:81]: why are these group rules and not global rules?

Comment (gurneyalex) [reply cms_security/utils.py:81]: my concern is that someone who is not a member of portal (e.g. an employee) will not be affected by the rules.

**Pull request closed without merging**